### PR TITLE
Milestone 3: Events

### PR DESCRIPTION
**Pull request by AkashaRojee**
Hello!

I implemented the event handlers to make the calculator work with the provided logic files.

Things done:

- Add provided logic files
- Implement click event handler for Key to register which calculator button is pressed
- Lift state up from Key to Calculator
- Set state to an object as required by calculate.js: `{total, next, operation}`
- Use custom object (DataObject) to store object
- Pass state as props to Display
- Construct string to display using values from data object
- Modify operate.js to cater for division by zero
- Modify calculate.js to cater for resetting display
- Fix linter errors
- Update README
- Deploy from branch

Please, review the code.

Thanks!

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
     super(props);
-    this.state = {};
+    this.state = {
+      dataObject: new DataObject(),
+    };
+  }
+
+  handleClick(key) {
+    this.setState((state) => ({
+      dataObject: { ...state.dataObject, ...calculate(state.dataObject, key) },
```
Comment (henatan99) [inline src/components/Calculator.js:17]: - [x] When an operator is clicked twice the app throws an error on the browser. Kindly incorporate error handling methods/conditions. 
image
- [x] When a number is divided by zero, your app returns a result which is `0`. But it should return some message like `Infinity`. Kindly handle this behavior.

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @AkashaRojee , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AkashaRojee** [comment]
Hi @henatan99,

Thanks for the feedback.

I implemented the changes.

Things done:

- Prevent operator buttons from being clicked consecutively

- Display "Undefined" as result of division by 0

Please, review the code.

Thanks!

**Review by henatan99 — APPROVED** [review]
Hi @AkashaRojee , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AkashaRojee** [comment]
Hi @henatan99,

Thanks!

**Pull request closed without merging**